src/cosim/observer/file_observer.cpp

```diff
@@ -301,10 +308,19 @@ class file_observer::slave_value_writer
     void persist()
     {
         std::stringstream ss;
+        const auto defaultPrecision = ss.precision();
+
         if (fsw_.is_open()) {
 
             for (const auto& [stepCount, times] : timeSamples_) {
-                ss << times << "," << stepCount;
+                if (precision_ > -1) {
+                    ss.precision(defaultPrecision);
+                    ss << times << "," << stepCount;
+                    ss.precision(precision_);
```

Comment:
Maybe a way to avoid setting precision twice pr. sample is to use two string writers, one for custom & one for default precision. Then at the end should be able to just go `fsw_ << ss_default.rdbuf() << ss_custom.rdbuf(); `

Reply:
Each line will have time & stepCount so don't think we can concatenate at once after the loop.